Framework/API/inc/MantidAPI/Result.h

```diff
@@ -0,0 +1,30 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2023 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include <string>
+
+namespace Mantid {
+namespace API {
+
+template <typename T> class MANTID_API_DLL Result {
+public:
+  Result(T result, std::string errors = "") : m_result(result), m_errors(errors) {}
```

Comment:
As there is a copy by the constructor you should be able to `move` the result and errors into the member variables:

```c++
 Result(T result, std::string errors = "") : m_result(std::move(result)), m_errors(std::move(errors)) {}
```